pkg/agent/client.go

```diff
@@ -243,7 +243,6 @@ func (a *Client) Send(pkt *client.Packet) error {
 	err := a.stream.Send(pkt)
 	if err != nil && err != io.EOF {
 		metrics.Metrics.ObserveFailure(metrics.DirectionToServer)
-		a.cs.RemoveClient(a.serverID)
```

Comment:
Not sure this line is safe to delete here. Send() is called from remoteToProxy() as well as Serve(). RemoveClient() is deliberately idempotent.

Reply:
Yeah I think you're right. We should reconnect the client immediately when sending bytes back up to the server returns an error. As it stands, I think this PR would defer reconnection until after the probe fails — assuming it does.

Reply:
Fixed.